script-library/common-alpine.sh

```diff
@@ -184,49 +184,66 @@ chmod +x /usr/local/bin/code
 
 # Codespaces themes - partly inspired by https://github.com/ohmyzsh/ohmyzsh/blob/master/themes/robbyrussell.zsh-theme
 CODESPACES_BASH="$(cat \
-<<EOF
-#!/usr/bin/env bash
-prompt() {
-    if [ "\$?" != "0" ]; then
-        local arrow_color=\${bold_red}
-    else
-        local arrow_color=\${reset_color}
-    fi
-    if [ ! -z "\${GITHUB_USER}" ]; then
-        local USERNAME="gh:@\${GITHUB_USER}"
-    else
-        local USERNAME="\$(whoami)"
-    fi
-    local cwd="\$(pwd | sed "s|^\${HOME}|~|")"
-    PS1="\${green}\${USERNAME} \${arrow_color}➜\${reset_color} \${bold_blue}\${cwd}\${reset_color} \$(scm_prompt_info)\${white}$ \${reset_color}"
+<<'EOF'
+__bash_prompt() {
+    local userpart='`export XIT=$? \
+        && [ ! -z "${GITHUB_USER}" ] && echo -n "\[\033[0;32m\]@${GITHUB_USER} " || echo -n "\[\033[0;32m\]\u " \
+        && [ "$XIT" -ne "0" ] && echo -n "\[\033[1;31m\]➜" || echo -n "\[\033[0m\]➜"`'
+    local gitbranch='`export BRANCH=$(git describe --contains --all HEAD 2>/dev/null); \
+        if [ "${BRANCH}" != "" ]; then \
+            echo -n "\[\033[0;36m\](\[\033[1;31m\]${BRANCH}" \
+            && if git ls-files --error-unmatch -m --directory --no-empty-directory -o --exclude-standard ":/*" > /dev/null 2>&1; then \
+                    echo -n " \[\033[1;33m\]✗"; \
+            fi \
+            && echo -n "\[\033[0;36m\]) "; \
+        fi`'
+    local lightblue='\[\033[1;34m\]'
+    local removecolor='\[\033[0m\]'
+    PS1="${userpart} ${lightblue}\w ${gitbranch}${removecolor}\$ "
+    unset -f __bash_prompt
 }
-SCM_THEME_PROMPT_PREFIX="\${reset_color}\${cyan}(\${bold_red}"
-SCM_THEME_PROMPT_SUFFIX="\${reset_color} "
-SCM_THEME_PROMPT_DIRTY=" \${bold_yellow}✗\${reset_color}\${cyan})"
-SCM_THEME_PROMPT_CLEAN="\${reset_color}\${cyan})"
-SCM_GIT_SHOW_MINIMAL_INFO="true"
-safe_append_prompt_command prompt
+__bash_prompt
 EOF
 )"
 CODESPACES_ZSH="$(cat \
-<<EOF
+<<'EOF'
 prompt() {
-    if [ ! -z "\${GITHUB_USER}" ]; then
-        local USERNAME="gh:@\${GITHUB_USER}"
+    if [ "${GITHUB_USER}" != "" ]; then
```

Comment:
Two questions:
1) I thought these were previously escaped `\$` instead of `$` because we don't want the variable evaluated when the file is written, but rather when it is executed by the user. We want to write the literal value `${GITHUB_USER}` into the generated script on disk. Does removing the escaping break that? It would be good to confirm that the generated file has the variable references intact.
2) Is there a functional difference with the string empty check versus `-z` for 0 length strings, or is that more of a style/preference change?

Reply:
Yeah on the escaping, the other thing I did was change this from `EOF` to `'EOF'` (note the ticks). `'EOF'` will not evaluate variables while `EOF` will.  Some of the other scripts take a variable from the main script - this one does not.

On `-z`, I probably could have left it alone... I must have done it at one point during edits. There's some [weirdness between unset and empty string when you're specifically looking for unset](https://mig8447.github.io/linux/bash/2018/07/25/checking-if-a-bash-variable-is-unset.html). But in this case, unset or empty string is actually what we want, so `-z` should be fine. I could revert it.

Reply:
@jkeech Actually, funnily enough, I noticed that code in the ZSH theme wasn't working.  Turns out, USERNAME doesn't work as a variable in this context so I switched it to a local variable called `prompt_username` and that fixed it.